### PR TITLE
Added Wrangler `npm run dev` command

### PR DESCRIPTION
**Pull request by geelen**
In `packages/wrangler`, run `npm run build-watch` to use esbuild's watch mode if you're hacking on stuff.

Builds on #1276 (stacked diffs when, Github??)

### CODE REVIEW

File: packages/wrangler/scripts/bundle.ts [hunk 1]
```diff
@@ -42,9 +55,29 @@ async function run() {
...
 }
 
+async function run() {
+  // main cli
+  await buildMain();
+
+  // custom miniflare cli
+  await buildMiniflareCLI();
+
+  console.log("Built wrangler & Miniflare CLI.");
```
Comment (threepointone) [inline packages/wrangler/scripts/bundle.ts:69]:
```suggestion
```

We shouldn't be saying we built the miniflare cli (we don't mention miniflare anywhere either), but let's not add this anyway?

File: packages/wrangler/package.json [hunk 2]
```diff
@@ -122,6 +122,7 @@
     "check:type": "tsc",
     "bundle": "node -r esbuild-register scripts/bundle.ts",
     "build": "npm run clean && npm run bundle",
+    "build-watch": "npm run clean && npm run bundle -- --watch",
```
Comment (threepointone) [inline packages/wrangler/package.json:125]:
```suggestion
    "watch": "npm run clean && npm run bundle -- --watch",
```

wanna make this a short word? it'll only be used by us, so let's make it easy

**Review by threepointone — APPROVED** [review]
Looking good! Dropped a couple of notes

File: packages/wrangler/package.json (hunk 2 above)
Comment (geelen) [reply packages/wrangler/package.json:125]: This was just to keep parity with `test-watch` below. Imo `:watch` should be a suffix you can apply to a bunch of commands. If we wanted to keep this short, maybe calling it `dev` would be better?

**Comment by geelen** [comment]
Changed this semantically to be `npm run dev`, which now includes TS checks alongside builds:

image

File: packages/wrangler/package.json [hunk 3]
```diff
@@ -40,6 +40,7 @@
     "@esbuild-plugins/node-globals-polyfill": "^0.1.1",
     "@esbuild-plugins/node-modules-polyfill": "^0.1.4",
     "blake3-wasm": "^2.1.5",
+    "concurrently": "^7.2.2",
```
Comment (threepointone) [inline packages/wrangler/package.json:43]: oops, this should be under devDependencies

**Pull request merged**